lib/core/async/connect.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+import 'dart:isolate';
+
+/// A function which is called as the entry point of a new isolate.
+typedef SpawnFunc = void Function(SendPort, ReceivePort);
+
+/// Data structure to hold an [Isolate] and
+/// it's send and receive ports.
+///
+/// The ports are either connected to the main [Isolate]
+/// or, if the class is used in the main [Isolate], to the
+/// spawned child [Isolate]s.
+class SpawnedIsolate {
+  /// The [Isolate] object.
+  final Isolate isolate;
+
+  /// The send port to the parent isolate.
+  ///
+  /// Should be initialized by user code when
+  /// the partner [Isolate] has send the
+  /// [SendPort] instance.
+  late final SendPort sendPort;
```

Comment:
Unfortunately it's necessary here. The send port is exchanged with the partner isolate. Therefore we do not have access to it until the partner isolate sends the port object to our receive port. On creation of `SpawnedIsolate` we did not receive this port because the messaging is managed by `Interface`. We cannot listen for the port and then pass control to `Interface` because `ReceivePort` is a single subscriber stream.

Using an optional is also not the way to go because we'd need to terminate the app anyway when we do not have a send port because it's absolutely critical for communication.

Reply:
There is `asBroadcastStream`, but that would probably only add unnecessary complexity

Reply:
Actually the `asBroadcastStream` does not work for whatever reason. I tried it :(